odoo/addons/base/models/ir_model.py

```diff
@@ -129,7 +129,14 @@ def upsert_en(model, fnames, rows, conflict):
     cols = ", ".join(quote(fname) for fname in fnames)
     values = ", ".join("%s" for row in rows)
     conf = ", ".join(conflict)
-    excluded = ", ".join(f"EXCLUDED.{quote(fname)}" for fname in fnames)
+    excluded = ", ".join(
+        (
+            f"COALESCE({table}.{quote(fname)}, '{{}}'::jsonb) || EXCLUDED.{quote(fname)}"
+            if model._fields[fname].translate
```

Comment:
```suggestion
            if model._fields[fname].translate is True
```
I prefer to explicitly claim the field.translate is True, since if we have `field.translate=html_trasnlate` in the future, these fields must be handled differently.

Reply:
When `field.translate=html_translate`, you cannot only update en_US, you must promise the html structures for all language translations are the same.
e.g. old translations `'{"en_US": "<p> term_en </p>", "fr_FR": "<p> term_fr </p>"}'::jsonb`
you cannot update it to `'{"en_US": "<p> term_en </p><p>term2_en</p>", "fr_FR": "<p> term_fr </p>"}'::jsonb` 
Otherwise we will break the data consistency for translated fields
It is not a problem for now, since all fields updated here don't have `field.translate=html_translate`. But semantically, this fix cannot handle `field.translate=html_translate`

Reply:
@HydrionBurst thanks, I updated